src/item.cpp

```diff
@@ -5838,13 +5838,12 @@ std::string item::get_plant_name() const
 
 bool item::is_dangerous() const
 {
-    // Note: doesn't handle the pipebomb or radio bombs
-    // Consider flagging dangerous items with an explicit flag instead
-    static const std::string explosion_string( "explosion" );
-    if( type->can_use( explosion_string ) ) {
+    if( has_flag( "DANGEROUS" ) ) {
         return true;
     }
 
+    // Note: Item should be dangerous regardless of what type of a container is it
+    // Visitable interface would skip some options
```

Comment:
Perhaps it should skip them? In what circumstances would that be a problem? Hidden `contents` are effectively incorporated in their parent item and shouldn't be directly accessed. If you need access to one of those hidden items `item::is_dangerous` should contain an explicit check?